tests-fuzz/src/generator/create_expr.rs

```diff
@@ -200,6 +206,116 @@ impl<R: Rng + 'static> Generator<CreateTableExpr, R> for CreateTableExprGenerato
+    name_generator: Box<dyn Random<Ident, R>>,
+}
+
+impl<R: Rng + 'static> Generator<CreateTableExpr, R> for CreatePhysicalTableExprGenerator<R> {
+    type Error = Error;
+
+    fn generate(&self, rng: &mut R) -> Result<CreateTableExpr> {
+        let if_not_exists = rng.gen_bool(0.5);
+
+        Ok(CreateTableExpr {
+            table_name: self.name_generator.gen(rng),
+            columns: vec![
+                Column {
+                    name: Ident::new("ts"),
+                    column_type: ConcreteDataType::timestamp_millisecond_datatype(),
+                    options: vec![ColumnOption::TimeIndex],
+                },
+                Column {
+                    name: Ident::new("val"),
+                    column_type: ConcreteDataType::float64_datatype(),
+                    options: vec![],
+                },
+            ],
+            if_not_exists,
+            partition: None,
+            engine: "metric".to_string(),
+            options: [("physical_metric_table".to_string(), "".into())].into(),
+            primary_keys: vec![],
+        })
+    }
+}
+
+/// Generate a logical table based on an existing physical table.
+#[derive(Builder)]
+#[builder(pattern = "owned")]
+pub struct CreateLogicalTableExprGenerator<R: Rng + 'static> {
+    physical_table_ctx: TableContextRef,
+    #[builder(default = "Box::new(WordGenerator)")]
+    name_generator: Box<dyn Random<Ident, R>>,
+    labels: usize,
+}
+
+impl<R: Rng + 'static> Generator<CreateTableExpr, R> for CreateLogicalTableExprGenerator<R> {
+    type Error = Error;
+
+    fn generate(&self, rng: &mut R) -> Result<CreateTableExpr> {
+        // Currently we mock the usage of GreptimeDB as Prometheus' backend, the physical table must have two columns.
+        ensure!(
+            self.physical_table_ctx.columns.len() == 2,
+            error::UnexpectedSnafu {
+                violated: "The physical table must have two columns"
+            }
+        );
+
+        // Generates the logical table columns based on the physical table.
+        let physical_table_name = self.physical_table_ctx.name.to_string().replace('`', "");
+        let table_generator = CreateTableExprGeneratorBuilder::default()
+            .name_generator(Box::new(MappedGenerator::new(
+                WordGenerator,
+                merge_two_word_map_fn(random_capitalize_map, uppercase_and_keyword_backtick_map),
+            )))
+            .columns(self.labels)
+            .engine("metric")
+            .column_type_generator(Box::new(StringColumnTypeGenerator))
+            .column_options_generator(Box::new(primary_key_column_options_generator))
+            .with_clause([("on_physical_table".to_string(), physical_table_name)])
+            .build()
+            .unwrap();
+
+        let mut table = table_generator.generate(rng)?;
+        table.table_name = self
+            .physical_table_ctx
+            .generate_unique_table_name(rng, self.name_generator.as_ref());
```

Comment:
I prefer using `generate_columns ` function to generate columns instead of `table_generator`, it be like:
https://github.com/GreptimeTeam/greptimedb/blob/9206f60b2828d2b759dad2f39780bcdbf739b092/tests-fuzz/src/generator/create_expr.rs#L167-L172